src/output.cpp

```diff
@@ -341,33 +341,76 @@ std::string Output::value_to_str(BPFtrace &bpftrace,
       return std::to_string(read_data<uint64_t>(value.data()) / div);
     }
     case Type::integer: {
-      auto sign = type.IsSigned();
-      switch (type.GetIntBitWidth()) {
-          // clang-format off
-        case 64:
-          if (sign)
-            return std::to_string(reduce_value<int64_t>(value, nvalues) / static_cast<int64_t>(div));
-          return std::to_string(reduce_value<uint64_t>(value, nvalues) / div);
-        case 32:
-          if (sign)
-            return std::to_string(
-                reduce_value<int32_t>(value, nvalues) / static_cast<int32_t>(div));
-          return std::to_string(reduce_value<uint32_t>(value, nvalues) / div);
-        case 16:
-          if (sign)
-            return std::to_string(
-                reduce_value<int16_t>(value, nvalues) / static_cast<int16_t>(div));
-          return std::to_string(reduce_value<uint16_t>(value, nvalues) / div);
-        case 8:
-          if (sign)
-            return std::to_string(
-                reduce_value<int8_t>(value, nvalues) / static_cast<int8_t>(div));
-          return std::to_string(reduce_value<uint8_t>(value, nvalues) / div);
-          // clang-format on
-        default:
-          LOG(BUG) << "value_to_str: Invalid int bitwidth: "
-                   << type.GetIntBitWidth() << "provided";
-          return {};
+      // Note we make the a conscious choice to only symbolize enums in
+      // map keys. The biggest concern is that it makes sense to do arith
+      // on enums sometimes. So if user does `SOME_ENUM + 1`, we could
```

Comment:
Hmm, I'm a little skeptical of this happening. Would be curious to see a real-world example but I won't block on it.

Reply:
Thinking again I don't feel strongly about it either way. Hard to think of a case where you'd keep an enum in value of map, though. It's like a ~2 line change either way.

Reply:
One edge case: when we enable configurable map types and the user chooses a percpu map, then symbolization makes no sense. If each CPU had a different enum val, what do we do?

Reply:
Exposing per-cpu maps to users I think poses some unique challenges we'll have to discuss.

We really don't have a good solution for how we would print any value that's kept in a percpu map that's not intended to be summed in some way (e.g. count, sum, avg). Maybe we just need to provide some utility that allows users to either print maps with the cpuid paired with the value (like a tuple) or to be able to iterate over cpus and print map values specific to that cpu.

Reply:
Good point. Percpu presents challenge for aggregation. So you're right it doesn't intersect with this PR. Let's symbolize map values, then. And continue discussion on percpu on the map type declarations issue.